ImageWrapper.cpp

```diff
@@ -0,0 +1,70 @@
+#include "lodepng.h"
+#include <vector>
+#include <iostream>
+#include "ImageWrapper.h"
+//#include "Log.h" idk why but this isnt working
+
+
+
+
+std::vector<unsigned char> ImageWrapper::decodeImageToRGBAVector(const char* filename) 
+{
+
+	std::vector<unsigned char> imagepixels;
+	 //the raw pixels, size: 8 bits, range: 0 -> 255
+	unsigned int width, height;
+
+	//decode
+	unsigned error = lodepng::decode(imagepixels, width, height, filename);
+
+	//if there's an error, display it
+	if (error) std::cout << "decoder error " << error << ": " << lodepng_error_text(error) << std::endl;
+
+	return imagepixels;
+}
+
+//R, G, B, A, R, G, B, A, R, G, B, A, R, G, B, A <- RGBAVector
+//increments by 4 through RGBAVector to fill out pixel struct, then adds to pixelVectorToReturn then returns it.
+//maybe some optimization here somehow through pointers/refs, i should definitely not pass
+//an entire copy of RGBAVector
+std::vector<pixel> ImageWrapper::convertRGBAVectorToPixelVector(std::vector<unsigned char>& RGBAVector)
```

Comment:
You're already passing the vector as a reference! perhaps just making it a const reference here is a good little optimization for now since we aren't changing / mutating the vector at all in this method

std::vector<pixel> ImageWrapper::convertRGBAVectorToPixelVector(const std::vector<unsigned char>& RGBAVector)